public/app/plugins/datasource/grafana-azure-monitor-datasource/components/MetricsQueryEditor/AggregationField.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback } from 'react';
+import { Select } from '@grafana/ui';
+
+import { Field } from '../Field';
+import { findOption } from '../common';
+import { SelectableValue } from '@grafana/data';
+import { AzureQueryEditorFieldProps, Option } from '../../types';
+
+interface AggregationFieldProps extends AzureQueryEditorFieldProps {
+  aggregationOptions: Option[];
+}
+
+const AggregationField: React.FC<AggregationFieldProps> = ({ query, onQueryChange, aggregationOptions }) => {
+  const handleChange = useCallback(
```

Comment:
not sure if I'm missing something here, but what's the real benefit of wrapping this function in useCallback?

Reply:
useCallback will memoize the function, so it won't 'change' on every render, potentially ca causing unnessisary renders.